OrbitClientModel/CaptureDeserializerTest.cpp

```diff
@@ -0,0 +1,408 @@
+
+  EXPECT_CALL(listener, OnCaptureStarted).Times(1);
+  EXPECT_CALL(listener, OnCaptureComplete).Times(1);
+
+  capture_deserializer::internal::LoadCaptureInfo(capture_info, &listener, &empty_stream,
+                                                  &cancellation_requested);
+}
+
+TEST(CaptureDeserializer, LoadCaptureInfoKeysAndStrings) {
+  MockCaptureListener listener;
+  CaptureInfo capture_info;
+
+  absl::flat_hash_map<uint64_t, std::string> keys_and_strings;
+  keys_and_strings[1] = "string_a";
+  keys_and_strings[2] = "string_b";
+
+  capture_info.mutable_key_to_string()->insert(keys_and_strings.begin(), keys_and_strings.end());
+  std::atomic<bool> cancellation_requested = false;
+  uint8_t empty_data = 0;
+  google::protobuf::io::CodedInputStream empty_stream(&empty_data, 0);
+
+  EXPECT_CALL(listener, OnKeyAndString(_, _)).Times(0);
+  EXPECT_CALL(listener, OnKeyAndString(1, "string_a")).Times(1);
+  EXPECT_CALL(listener, OnKeyAndString(2, "string_b")).Times(1);
+
+  EXPECT_CALL(listener, OnCaptureStarted).Times(1);
+  EXPECT_CALL(listener, OnCaptureComplete).Times(1);
+
+  capture_deserializer::internal::LoadCaptureInfo(capture_info, &listener, &empty_stream,
+                                                  &cancellation_requested);
+}
+
+TEST(CaptureDeserializer, LoadCaptureInfoCallstacks) {
+  MockCaptureListener listener;
+  CaptureInfo capture_info;
+
+  // Add two callstacks with same hash:
+  std::vector<uint64_t> callstack_data_1;
+  callstack_data_1.push_back(1);
+  callstack_data_1.push_back(2);
+  callstack_data_1.push_back(3);
+  CallStack callstack_1(std::move(callstack_data_1));
+  CallstackInfo* callstack_info_1 = capture_info.add_callstacks();
+  *callstack_info_1->mutable_data() = {callstack_1.GetFrames().begin(),
+                                       callstack_1.GetFrames().end()};
+  CallstackEvent* callstack_event_1_1 = capture_info.add_callstack_events();
+  callstack_event_1_1->set_thread_id(1);
+  callstack_event_1_1->set_time(1);
+  callstack_event_1_1->set_callstack_hash(callstack_1.GetHash());
+  CallstackEvent* callstack_event_1_2 = capture_info.add_callstack_events();
+  callstack_event_1_2->set_thread_id(1);
+  callstack_event_1_2->set_time(2);
+  callstack_event_1_2->set_callstack_hash(callstack_1.GetHash());
+
+  // Add one additional callstack with a different hash:
+  std::vector<uint64_t> callstack_data_2;
+  callstack_data_2.push_back(4);
+  callstack_data_2.push_back(5);
+  CallStack callstack_2(std::move(callstack_data_2));
+  CallstackInfo* callstack_info_2 = capture_info.add_callstacks();
+  *callstack_info_2->mutable_data() = {callstack_2.GetFrames().begin(),
+                                       callstack_2.GetFrames().end()};
+  CallstackEvent* callstack_event_2 = capture_info.add_callstack_events();
+  callstack_event_2->set_thread_id(2);
+  callstack_event_2->set_time(3);
+  callstack_event_2->set_callstack_hash(callstack_1.GetHash());
```

Comment:
Did you mean `callstack_2`?
```suggestion
  callstack_event_2->set_callstack_hash(callstack_2.GetHash());
```

Reply:
Thanks!